### PR TITLE
copr: fix wrong json opaque serialization

### LINKED ISSUE #13391: Make the serialization of opaque value compatible with MySQL
## Development Task

I have brought in the ability to serialize opaque value in https://github.com/tikv/tikv/pull/13342. However, the format is not compatible with MySQL, can cause https://github.com/pingcap/tidb/issues/37501.

### PR DESCRIPTION
**Pull request by YangKeao**
Signed-off-by: YangKeao <yangkeao@chunibyo.icu>

### What is changed and how it works?

Issue Number: Close #13391 

What's Changed:

1. Change the format of json opaque serialization
2. Add more tests about it

### Check List 

Tests 

- [x] Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

### Release note 

```release-note
Fix the issue that the serialization of opaque value is not compatible with MySQL
```

### CODE REVIEW

**Comment by YangKeao** [comment]
@xiongjiwei PTAL

File: components/tidb_query_expr/src/impl_json.rs [hunk 1]
```diff
@@ -213,7 +213,10 @@ fn json_unquote(arg: BytesRef) -> Result<Option<Bytes>> {
     if tmp_str.len() >= 2 && first_char == Some('"') && last_char == Some('"') {
         let _: IgnoredAny = serde_json::from_str(tmp_str)?;
     }
-    Ok(Some(Bytes::from(self::unquote_string(tmp_str)?)))
+
+    let ret = self::unquote_string(tmp_str)?;
+    println!("UNQUOTE: {:?}", ret);
```
Comment (xiongjiwei) [inline components/tidb_query_expr/src/impl_json.rs:218]: println?

**Review by xiongjiwei — APPROVED** [review]

File: components/tidb_query_expr/src/impl_json.rs (hunk 1 above)
Comment (YangKeao) [reply components/tidb_query_expr/src/impl_json.rs]: Nice catch, removed :facepalm:

**Review by xhebox — APPROVED** [review]

**Review by sticnarf — APPROVED** [review]
LGTM

**Comment by YangKeao** [comment]
@gengliqi @wshwsh12 PTAL, a small bugfix :heart:

**Review by gengliqi — APPROVED** [review]

**Pull request merged**